### PR TITLE
Problem: ibc channel upgrade related methods are not supported

### PR DESCRIPTION
**Pull request by mmsqe**

### CODE REVIEW

**Review by yihuang — APPROVED** [review]

**Comment by mmsqe** [comment]
strange that ci is green in my [action](https://github.com/mmsqe/pystarport/actions/runs/11319841440/job/31476494216), any clue? @yihuang

**Comment by yihuang** [comment]
> strange that ci is green in my [action](https://github.com/mmsqe/pystarport/actions/runs/11319841440/job/31476494216), any clue? @yihuang

try this: https://github.com/crypto-com/pystarport/pull/140

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,18 +13,19 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: cachix/install-nix-action@v30
-      - uses: DeterminateSystems/magic-nix-cache-action@main
+      - uses: cachix/install-nix-action@v15
```
Comment (yihuang) [inline .github/workflows/test.yml:16]:
```suggestion
      - uses: cachix/install-nix-action@v30
      - uses: DeterminateSystems/magic-nix-cache-action@main
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -21,10 +21,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: cachix/install-nix-action@v30
-      - uses: DeterminateSystems/magic-nix-cache-action@main
+      - name: install dependencies
+        run: python3 -m pip install --user --upgrade poetry
+      - name: install
+        run: poetry install
```
Comment (yihuang) [inline .github/workflows/test.yml:27]:
```suggestion
      - uses: cachix/install-nix-action@v30
      - uses: DeterminateSystems/magic-nix-cache-action@main
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -24,7 +24,7 @@ jobs:
       - uses: cachix/install-nix-action@v30
       - uses: DeterminateSystems/magic-nix-cache-action@main
       - name: Run tests
-        run: nix develop -c python -mpytest -vv -s
+        run: make test
```
Comment (yihuang) [inline .github/workflows/test.yml:27]:
```suggestion
        run: nix develop -c python -mpytest -vv -s
```

**Pull request merged**